### DISCUSSION TITLE
Way to get boxscores of just one specified team Basketball

### DISCUSSION THREAD

**Discussion by smoudour** [cwendt94/espn-api#480] [ESPN API]
Hi, I've been developing an extension of the API concerning NBA Fantasy Stats and Analyses. I was wondering if there is a more efficient way to get the scores of a specific team in a 9 cat league for each week's matchup, other than getting all the scores for each week (using `box_score` member function and then looping through all the acquired matchups. It would be faster to just grab the matchup of interest (i.e. the one containing the specified team only).

**Comment by cwendt94** [comment]
Are you only looking for the score for each week in that matchup? We might be able to get that data from the `schedule` filed in each [Team Class](https://github.com/cwendt94/espn-api/wiki/Team-Class-Basketball). It is using the Matchup class but Im not sure how it would look for 9 cat league. But that could be a starting point and then update the logic for cat leagues.

**Reply by smoudour** [reply]
I'm looking for the values in each category and not just the overall score (e.g. 6-3). What I'm trying to do is for example compare stats of team 1 and team 2 for every week, like if they played against each other every time. Currently I'm getting the complete boxscore for every week, and then using something like: 

```
def boxscores_to_dict(week_boxscores: list[H2HCategoryBoxScore]) -> dict:

    teams_stats = {}
    for boxscore in week_boxscores:
        team_name_1 = boxscore.home_team.team_name
        team_name_2 = boxscore.away_team.team_name
        teams_stats[team_name_1] = boxscore.home_stats
        teams_stats[team_name_2] = boxscore.away_stats

    return teams_stats
```
and getting all team stats for say week 1, then isolating stats of desired team(s). Thing is that when getting the `week_boxscores` list, using `my_league.box_scores(week_x)` I am unnecessarily getting information for every team, while I only need stats for team1 and team2 - process is slow when doing this for the whole season. 
Basically what I am asking is if there is a way to acquire stats (category values) of selected team for selected week directly through my_leage, bypassing boxscores.

**Reply by cwendt94** [reply]
Okay, yeah I don't think the matchup would help you here. I will have to look to see if there is a way to get all of that data for a specific team.
